strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/domain/ArtifactEntry.java

```diff
@@ -51,6 +45,9 @@
 
     private Integer downloadCount = Integer.valueOf(0);
 
+    @Transient
+    private Set<ArtifactGroup> artifactGroups;
```

Comment:
I found that this `Transient` field really confusing me. Can we drop it?

Reply:
This way we pass the `artifactGroups` to the service layer and we can save artifact entries related with it (see [here](https://github.com/strongbox/strongbox/pull/987/files#diff-ca2af8f423f4756b1737ad2178c1101dR82)). Otherwise we would have to pass it as an additional parameter to the `ArtifactEntryServiceImpl#save(S, boolean)` method (keeping the `artifactGroups`  transactional with `ArtifactEntry`).

Reply:
I think that we probably need to have `ArtifactGroupService.addArtifactToGroup` method, and move `last-version` tag update logic there.

> keeping the artifactGroups transactional with ArtifactEntry

we have such transaction in `ArtifactManagementService`

Reply:
I got rid of this field.